distributed/scheduler.py

```diff
@@ -4556,20 +4567,25 @@ def _add_to_memory(
             msg = {"op": "key-in-memory", "key": ts._key}
             if type is not None:
                 msg["type"] = type
-            self.report(msg)
 
         ts.state = "memory"
         ts._type = typename
         ts._group._types.add(typename)
 
         cs: ClientState = self.clients["fire-and-forget"]
         if ts in cs._wants_what:
-            self.client_releases_keys(client="fire-and-forget", keys=[ts._key])
+            recommendations.update(
+                self._client_releases_keys(client="fire-and-forget", keys=[ts._key])
+            )
+
+        return msg
 
     def transition_released_waiting(self, key):
         try:
             ts: TaskState = self.tasks[key]
             dts: TaskState
+            worker_msgs: dict = {}
+            report_msg: dict = {}
```

Comment:
Should these be `None` for the seemingly common case of no messages to report?

Reply:
Well as we have `for`-loops back in `transition` that try to iterate over these, we would need to add logic to check for `None` somehow. Atm this seems pretty straightforward even if there are cases where one of these doesn't contain something.

Reply:
Should add that ultimately my hope is this becomes a message object, which we pass from transition functions and the communication side of the Scheduler. For now it is sort of an arbitrary list of things we have found need to be passed around. So I think now that we have identified what those things that we message are, this should be easier to handle.